### PR TITLE
feat(witness): use block executor to execute block inside debug_execution_witness

### PR DESCRIPTION
**Pull request by 0x00101010**
The code does the following:

1. allow passing BlockExecutorProvider to RpcModuleBuilder and eventually passed down to DebugApi initialization
2. rework debug_execution_witness API to use block_executor to execute blocks.

### CODE REVIEW

**Comment by mattsse** [comment]
cool this should actually do it!
we do this first and then make it nicer by finding a better approach

File: crates/rpc/rpc/src/debug.rs [hunk 1]
```diff
@@ -44,16 +43,22 @@ use tokio::sync::{AcquireError, OwnedSemaphorePermit};
 /// `debug` API implementation.
 ///
 /// This type provides the functionality for handling `debug` related requests.
-pub struct DebugApi<Provider, Eth> {
-    inner: Arc<DebugApiInner<Provider, Eth>>,
+pub struct DebugApi<Provider, Eth, BlockExecutor> {
+    inner: Arc<DebugApiInner<Provider, Eth, BlockExecutor>>,
```
Comment (mattsse) [inline crates/rpc/rpc/src/debug.rs:47]: @Rjected as discussed we can move this type in here first and in later step unify execution step by step.

File: crates/rpc/rpc/src/debug.rs [hunk 2]
```diff
@@ -572,107 +578,159 @@ where
...
-                            let slot = B256::from(slot);
-                            let hashed_slot = keccak256(slot);
-                            storage.storage.insert(hashed_slot, value);
-
-                            if include_preimages {
-                                state_preimages.insert(hashed_slot, alloy_rlp::encode(slot).into());
+                let mut keys = HashMap::new();
+                let _ = block_executor
+                    .execute_with_state_witness(
+                        (&block.clone().unseal(), block.difficulty).into(),
```
Comment (mattsse) [inline crates/rpc/rpc/src/debug.rs:599]: you can get rid of this clone if you unseal into a tmp var

File: crates/rpc/rpc/src/debug.rs [hunk 3]
```diff
@@ -572,107 +578,159 @@ where
...
+                // // apply eip-2935 blockhashes update
+                // pre_block_blockhashes_contract_call(
+                //     &mut db,
+                //     &evm_config,
+                //     &this.inner.provider.chain_spec(),
+                //     &cfg,
+                //     &block_env,
+                //     block.parent_hash,
+                // )
+                // .map_err(|err| EthApiError::Internal(err.into()))?;
```
Comment (mattsse) [inline crates/rpc/rpc/src/debug.rs:665]: cool all of this we can remove now

**Review by mattsse — CHANGES_REQUESTED** [review]
awesome, this is great, exactly what I had in mind as the intermediary solution.

just needs doc cleanup

**Review by mattsse — APPROVED** [review]
nice,
the state hook looks correct to, mind doing a sanity check @Rjected

**Review by Rjected — APPROVED** [review]
nice! this looks good to me

**Pull request merged**